### PR TITLE
Filter for RecordDataBase

### PR DESCRIPTION
**Pull request by nickhaf**
Implemented a filter function for databases.
Wrote a method for `Base.filter()` that takes a filter function, a database and, additionally, the level on which the filter function should be applied (Record, Study, Model, Taxon). 
The filter function can also work with `rating()` or `certainty()` so the function definition works a bit more naturally, see documentation of `filter()`.

### CODE REVIEW

File: src/filter.jl [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+
+
+# output
+1-element Vector{Union{Study, AbstractJudgement}}:
+ Study(Dict{Symbol, Vector{Union{JudgementLevel, AbstractJudgement}}}(:N => [N{Int64}(200, 0.98, missing)]))
+```
+"""
+function Base.filter(f, db::RecordDatabase, level::AbstractString)::RecordDatabase
+
+    @assert level in ["Record", "Study", "Model", "Taxon"] "Invalid level. Choose from 'Record', 'Study', 'Model', 'Taxon'"
```
Comment (nickhaf) [inline src/filter.jl:102]: Not quite happy with asserting the filter level like that, do you have a more elegant way?

Comment (nickhaf) [inline src/filter.jl:102]: Not that happy with this solution, works for now but do you have a more elegant way?

**Review by aaronpeikert — APPROVED** [review]
This is a much needed feature. Thank you for taking it on! I love that you invested so much in testing.

**Comment by aaronpeikert** [comment]
Ignore the failure on nightly.

**Pull request merged**